src/region/__tests__/RegionAnnotations-test.tsx

```diff
@@ -1,95 +1,25 @@
-import RegionRect from '../RegionRect';
 import { annotations } from '../__mocks__/data';
-import { CreatorItem, CreatorStatus } from '../../store';
-import { Rect } from '../../@types';
 
-jest.mock('../../components/Popups/PopupReply');
-jest.mock('../RegionCreator');
 jest.mock('../RegionList');
-jest.mock('../RegionRect');
 
 describe('RegionAnnotations', () => {
     const defaults = {
         activeAnnotationId: null,
-        location: 1,
-        resetCreator: jest.fn(),
+        annotations: [],
         setActiveAnnotationId: jest.fn(),
-        setMessage: jest.fn(),
-        setReferenceShape: jest.fn(),
-        setStaged: jest.fn(),
-        setStatus: jest.fn(),
-        staged: null,
     };
-    const getRect = (): Rect => ({
-        type: 'rect',
-        height: 50,
-        width: 50,
-        x: 10,
-        y: 10,
-    });
-    const getStaged = (): CreatorItem => ({
-        location: 1,
-        shape: getRect(),
-    });
     const getWrapper = (props = {}): ShallowWrapper => shallow(<RegionAnnotations {...defaults} {...props} />);
 
     describe('event handlers', () => {
-        let wrapper: ShallowWrapper;
-        let instance: InstanceType<typeof RegionAnnotations>;
-
-        beforeEach(() => {
-            wrapper = getWrapper({
-                staged: getStaged(),
-            });
-            instance = wrapper.instance() as InstanceType<typeof RegionAnnotations>;
-        });
-
-        describe('handleAbort', () => {
-            test('should call resetCreator', () => {
-                instance.handleAbort();
-
-                expect(defaults.resetCreator).toHaveBeenCalled();
-            });
-        });
-
-        describe('handleStart', () => {
-            test('should reset the creator status', () => {
-                instance.handleStart();
-
-                expect(defaults.setStaged).toHaveBeenCalledWith(null);
-                expect(defaults.setStatus).toHaveBeenCalledWith(CreatorStatus.started);
-            });
-        });
-
-        describe('handleStop', () => {
-            const shape = {
-                type: 'rect' as const,
-                height: 10,
-                width: 10,
-                x: 20,
-                y: 20,
-            };
-
-            test('should update the staged status and item with the new shape', () => {
-                instance.handleStop(shape);
-
-                expect(defaults.setStaged).toHaveBeenCalledWith({
-                    location: defaults.location,
-                    shape,
-                });
-                expect(defaults.setStatus).toHaveBeenCalledWith(CreatorStatus.staged);
-            });
-        });
-
         describe('handleAnnotationActive()', () => {
             test('should call setActiveAnnotationId with annotation id', () => {
-                instance.handleAnnotationActive('123');
+                (getWrapper()
+                    .find(RegionList)
+                    .prop('onSelect') as Function)('123');
```

Comment:
Can we avoid these casts?

Reply:
This is needed because `onSelect` is an optional prop